server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -285,9 +286,7 @@ static boolean executeBulkItemRequest(
                 context.markAsCompleted(context.getExecutionResult());
                 return true;
             }
-            DocWriteRequest<?> translated = updateResult.action();
-            translated.process();
-            context.setRequestToExecute(translated);
+            context.setRequestToExecute(updateResult.action());
```

Comment:
This change is interesting. `process` doesn't have the same signature so I had to change it *somehow*. It was a little complex to route all the *stuff* into the right spot so I thought "what happens if I just never call this - all `process` does right now is set the `id`. And we set the `id` already when building the action in the update stuff. It's probably ok to skip it." And so I tried it. And all the tests pass.....

Reply:
Which probably implies that `process` ain't a good name for this method. But I don't know of a good one.